spec/services/bookings/gitis/school_experience_spec.rb

```diff
@@ -0,0 +1,181 @@
+require 'rails_helper'
+
+describe Bookings::Gitis::SchoolExperience, type: :model do
+  describe "validations" do
+    let(:model) { create(:placement_request, :booked) }
+    let(:status) { :requested }
+    subject { described_class.from_placement_request(model, status) }
+
+    before { allow_any_instance_of(GetIntoTeachingApiClient::SchoolsExperienceApi).to receive(:add_school_experience) }
+
+    context "with valid arguments" do
+      it "does not throw error" do
+        expect { subject }.to_not raise_error
+      end
+    end
+
+    describe "#urn" do
+      it "is too long" do
```

Comment:
This will read a bit funny in the rspec output, something like:

```
validations #urn is too long
```

Can we make it more standard, along the lines of:

```
validations #urn is invalid when too long
```

Is there a reason you've not used shoulda matchers here? I think it would make it more concise?

Reply:
Shoulda matchers aren't working for me because I'm not using attributes. I wanted to make sure a `SchoolExperience` can only be created through the factory methods, and I don't think I can have private attributes.